lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb

```diff
@@ -1032,12 +1033,14 @@ def header_mappings_dir(file_accessor)
                                                    end
           end
 
-          def add_header(file_accessor, build_file, public_headers, private_headers, native_target)
+          def add_header(file_accessor, build_file, public_headers, project_headers, private_headers, native_target)
             file_ref = build_file.file_ref
             acl = if !target.build_as_framework? # Headers are already rooted at ${PODS_ROOT}/Headers/P*/[pod]/...
                     'Project'
                   elsif public_headers.include?(file_ref.real_path)
                     'Public'
+                  elsif project_headers.include?(file_ref.real_path)
+                    'Project'
```

Comment:
there is a check right below if you expand this file that includes:

```
if target.build_as_framework? && !header_mappings_dir(file_accessor).nil? && acl != 'Project'
```

This will ensure 'Project` headers _are not_ copied therefore could not be imported by other pods. Will double check this is true in my example.